modules/videoio/src/cap_msmf.cpp

```diff
@@ -1204,8 +1627,147 @@ bool CvCapture_MSMF::retrieveFrame(int, cv::OutputArray frame)
+        if (lastFrame && !syncLastFrame)
+        {
+            numberOfByte = (int)bufferAudioData.size();
+        }
+        if (numberOfByte % 2 != 0)
+            numberOfByte++;
+        useAudioData = new BYTE[numberOfByte];
+        for (int i = 0; i < numberOfByte; i++)
+        {
+            useAudioData[i] = bufferAudioData[i];
+        }
+        bufferAudioData.erase(bufferAudioData.begin(), bufferAudioData.begin()+numberOfByte);
+        residualTime = (double)(bufferAudioData.size()/((captureAudioFormat.bit_per_sample/8)*captureAudioFormat.nChannels))/captureAudioFormat.nSamplesPerSec;
+        audioSamplePos += numberOfByte/((captureAudioFormat.bit_per_sample/8)*captureAudioFormat.nChannels);
+        if (audioFrame.empty())
+        {
+            switch (outputAudioFormat)
+            {
+            case CV_8S:
+                cv::Mat(numberOfByte/(captureAudioFormat.nChannels), captureAudioFormat.nChannels, CV_8S, useAudioData).copyTo(audioFrame);
+                break;
+            case CV_16S:
+                cv::Mat(numberOfByte/(2*captureAudioFormat.nChannels), captureAudioFormat.nChannels, CV_16S, useAudioData).copyTo(audioFrame);
+                break;
+            case CV_32S:
+                cv::Mat(numberOfByte/(4*captureAudioFormat.nChannels), captureAudioFormat.nChannels, CV_32S, useAudioData).copyTo(audioFrame);
+                break;
+            case CV_32F:
+                cv::Mat(numberOfByte/(4*captureAudioFormat.nChannels), captureAudioFormat.nChannels, CV_32F, useAudioData).copyTo(audioFrame);
+                break;
+            default:
+                break;
+            }
+        }
+        cv::Mat data;
+        switch (outputAudioFormat)
+        {
+        case CV_8S:
+            data = cv::Mat(1, audioFrame.rows, CV_8S);
+            for (int i = 0; i < audioFrame.rows; i++)
+                data.at<char>(0,i) = audioFrame.at<char>(i,index-audioBaseIndex);
+            break;
+        case CV_16S:
+            data = cv::Mat(1, audioFrame.rows, CV_16S);
+            for (int i = 0; i < audioFrame.rows; i++)
+                data.at<short>(0,i) = audioFrame.at<short>(i,index-audioBaseIndex);
+            break;
+        case CV_32S:
+            data = cv::Mat(1, audioFrame.rows, CV_32S);
+            for (int i = 0; i < audioFrame.rows; i++)
+                data.at<int>(0,i) = audioFrame.at<int>(i,index-audioBaseIndex);
+            break;
+        case CV_32F:
+            data = cv::Mat(1, audioFrame.rows, CV_32F);
+            for (int i = 0; i < audioFrame.rows; i++)
+                data.at<float>(0,i) = audioFrame.at<float>(i,index-audioBaseIndex);
+            break;
+        default:
+            frame.release();
+            break;
+        }
+        if (!data.empty())
+            data.copyTo(frame);
+        delete[] useAudioData;
+    return !frame.empty();
+    } while (0);
```

Comment:
broken indentation

Reply:
> broken indentation

What do you mean by that?

Reply:
`return` statement needs plus 4 spaces in prefix